File: July 2024 GFG SOLUTION/July-29.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    int rowWithMax1s(vector<vector<int>>& arr) {
        int n = arr.size();       
        int m = arr[0].size();   
        int max_row_index = -1;   
        
        int r = 0;    
        int c = m - 1; 
        
        while (r < n && c >= 0) {
            if (arr[r][c] == 1) {
                max_row_index = r; 
                c--;              
            } else {
                r++;             
            }
        }
        
        return max_row_index;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector<vector<int> > arr(n, vector<int>(m));
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                cin >> arr[i][j];
            }
        }
        Solution ob;
        auto ans = ob.rowWithMax1s(arr);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: July 2024 GFG SOLUTION/July-29.py
#User function Template for python3
class Solution:
    def rowWithMax1s(self, arr):
        n = len(arr)        
        m = len(arr[0])       
        max_row_index = -1    
        
        r = 0   
        c = m - 1 
        
        while r < n and c >= 0:
            if arr[r][c] == 1:
                max_row_index = r 
                c -= 1            
            else:
                r += 1            
        
        return max_row_index

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n, m = list(map(int, input().strip().split()))

        inputLine = list(map(int, input().strip().split()))
        arr = [[0 for j in range(m)] for i in range(n)]

        for i in range(n):
            for j in range(m):
                arr[i][j] = inputLine[i * m + j]
        ob = Solution()
        ans = ob.rowWithMax1s(arr)
        print(ans)
        tc -= 1

# } Driver Code Ends
